src/pinnable_mapped_file.cpp

```diff
@@ -16,7 +16,8 @@
 namespace chainbase {
 
 std::vector<pinnable_mapped_file*> pinnable_mapped_file::_instance_tracker;
-
+pinnable_mapped_file::segment_manager_map_t  pinnable_mapped_file::_segment_manager_map;
```

Comment:
Consider updating
https://github.com/AntelopeIO/chainbase/blob/61ea098a8059581c279543b96976d32f4e031a6f/README.md?plain=1#L123-L124
since this statement is now true for any mode

Reply:
Can you elaborate as to why it is true in all modes? I thought it was due to the fact that we clear the `Soft-Dirty` bit for all instances when loading a new db, so it would not be the case for other modes. What am I missing?

Reply:
`_segment_manager_map` is modified in both the ctor & dtor no matter what mode,
https://github.com/AntelopeIO/chainbase/blob/c1758f709fd0d133a56fa01de9e000ca1d533b1a/src/pinnable_mapped_file.cpp#L227
https://github.com/AntelopeIO/chainbase/blob/c1758f709fd0d133a56fa01de9e000ca1d533b1a/src/pinnable_mapped_file.cpp#L457
But anytime anything calls `get_allocator()` an access on that map occurs,
https://github.com/AntelopeIO/chainbase/blob/c1758f709fd0d133a56fa01de9e000ca1d533b1a/include/chainbase/pinnable_mapped_file.hpp#L70-L72

So if a `pinnable_mapped_file` is being ctored or dtored on one thread at the same time another thread is calling `get_allocator()` (for a different `pinnable_mapped_file`), it's a violation of threading rules.

Reply:
Thanks for the explanation, Matt, got it!

Reply:
Updated the comment.